internal/rbd/group/util.go

```diff
@@ -85,7 +85,9 @@ func (cvg *commonVolumeGroup) generateVolumeGroup(csiID util.CSIIdentifier) erro
 	cvg.namespace = namespace
 	cvg.pool = pool
 
-	return nil
+	_, err = cvg.Exists(context.Background())
```

Comment:
Don't call this here, call it at a location that wants to check if the volume group exists.

Reply:
> Don't call this here, call it at a location that wants to check if the volume group exists.

@iPraveenParihar @nixpanic 
I don't think we need to have a function Exists at all. Its called at only two places, that too only after generateVolGroup call 

https://github.com/ceph/ceph-csi/blob/9eb4f61d015696171986f6167a512658d09aec2c/internal/rbd/group/util.go#L475